### ISSUE TITLE
Regex with braces as delimiters do not properly begin and end

### ISSUE THREAD

**Issue by zoffixznet** [perl6/atom-language-perl6#60]
I'm using 1.8.0

### Example Code
```perl6
    if $parent {
        while $path ~~ s:g { [^ | <?after '\\'>] <!before '..\\'> <-[\\]>+ '\\..' ['\\' | $ ] } = '' { };
    }
    $path ~~ s/^ '\\'+ //;        # \xx --> xx  NOTE: this is *not* root
```
### Picture [optional]
![untitled](https://cloud.githubusercontent.com/assets/5747918/25060278/45e1b368-2167-11e7-9ba3-f074f7492fd0.png)

### Leave this in. For internal use.
- [x] Fixed in Master
- [x] Fixed in Release
- [x] Has Tests
- [x] Passes Tests

**Comment by zoffixznet** [comment]
After playing with code, the something is the braces on s:g//, changing them to some other delimiter fixes the issue

**Comment by zoffixznet** [comment]
Oops. I was wrong. It only unbreaks when using something like `♥`, but that makes invalid code since we need matching braces in the `s{...} = 'something'` form

**Comment by zoffixznet** [comment]
I think I found it now, something to do with all the backslashes. This isn't broken:

```perl6
        if $parent {
            while $path ~~ s:g { [^ | <?after ｢\｣>] <!before ｢..\｣> <-[\\]>+ '\\..' ['\\' | $ ] } = '' { };
        }
```

**Comment by samcv** [comment]
This is a duplicate of https://github.com/perl6/atom-language-perl6/issues/56 but i'm going to keep this one open and close the other one.

The problem is that `s:g{ }` is not highlighting as regex **period**, and so this wrecks things when it doesn't know it's inside a regex. IF you change them to s:g /  / then it all is fine.

**Comment by samcv** [comment]
Fixed in 1.15.0 which has just been released. :heavy_check_mark:
